### PR TITLE
fix: use require.NoError instead of t.Fatal(err) in tests and tools packages

### PR DESCRIPTION
**Pull request by mmorel-35**
There is no linter for this.
This uses testify instead of testing for
With 
```go
require.NoError(t, err)

require.Error(t, err)
```
instead of

```go
if err != nil {
    t.Fatal(err)
}


if err == nil {
    t.Fatal(err)
}
```

This is only a part of this packages. I'll provide other PRs for the rest of it.

### CODE REVIEW

**Comment by ivanvc** [comment]
Hi @mmorel-35, PTAL at the prow job pull-etcd-verify failure: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/etcd-io_etcd/18709/pull-etcd-verify/1844226305705381888#1:build-log.txt%3A841

File: tests/common/alarm_test.go [hunk 1]
```diff
@@ -39,9 +41,7 @@ func TestAlarm(t *testing.T) {
 	testutils.ExecuteUntil(ctx, t, func() {
 		// test small put still works
 		smallbuf := strings.Repeat("a", 64)
-		if err := cc.Put(ctx, "1st_test", smallbuf, config.PutOptions{}); err != nil {
-			t.Fatalf("alarmTest: put kv error (%v)", err)
-		}
+		require.NoErrorf(t, cc.Put(ctx, "1st_test", smallbuf, config.PutOptions{}), "alarmTest: put kv error")
```
Comment (ahrtr) [inline tests/common/alarm_test.go:44]: minor comment
```suggestion
		err := cc.Put(ctx, "1st_test", smallbuf, config.PutOptions{})
		require.NoErrorf(t, err, "alarmTest: put kv error")
```

**Review by ahrtr — APPROVED** [review]
LGTM with a minor comment

Thanks

**Comment by ahrtr** [comment]
> LGTM with a minor comment
> 
> Thanks

Oops, approved the wrong PR.

**Pull request closed without merging**